runtime/compiler/z/codegen/S390J9CallSnippet.cpp

```diff
@@ -1718,3 +1720,183 @@ TR::S390InterfaceCallReadOnlySnippet::getLength(int32_t estimatedSnippetStart)
            + sizeof(int32_t)  /*RIP offset to PICSlot data address*/
            + sizeof(int32_t);  /*RIP offset to instruction in mainline*/
    }
+ 
+uint8_t *
+TR::S390J9CallSnippetRX::emitSnippetBody()
+   {
+   uint8_t *cursor = cg()->getBinaryBufferCursor();
+   TR::SymbolReference *callSymRef = getRealMethodSymbolReference();
+   TR::Node *callNode = getNode();
+   getSnippetLabel()->setCodeLocation(cursor);
+   cursor = S390flushArgumentsToStack(cursor, getNode(), getSizeOfArguments(), cg());
+
+   //Check if the call is unresolved
+   if (callSymRef->isUnresolved() || (cg()->comp()->compileRelocatableCode() && !cg()->comp()->getOption(TR_UseSymbolValidationManager)))
+      {
+      typedef J9::Z::PrivateLinkage::ccUnresolvedStaticSpecialData ccUnresolvedStaticSpecialData;
+      ccUnresolvedStaticSpecialData *ccDataAddress = reinterpret_cast<ccUnresolvedStaticSpecialData *>(callSnippetCCDataAddress);
+      ccDataAddress->snippetOrCompiledMethod = reinterpret_cast<intptr_t>(getSnippetLabel()->getCodeLocation());
+      // LGRL  rEP, @(CCUnresolvedStaticOrSpecialData + ramMethod)
+      intptr_t instrAddr = reinterpret_cast<intptr_t>(cursor);
+      *cursor = static_cast<uint8_t>(0xC4);
+      cursor += sizeof(uint8_t);
+      *reinterpret_cast<int8_t *>(cursor) = static_cast<int8_t>(((cg()->getEntryPointRegister() - 1) << 4 ) + 0x8 );
+      cursor += sizeof(int8_t);
+      *reinterpret_cast<int32_t *>(cursor) = static_cast<int32_t>((callSnippetCCDataAddress + offsetof(ccUnresolvedStaticSpecialData, ramMethod) - instrAddr) / 2 );
+      cursor += sizeof(int32_t);
+
+      // CGIJ rEP,0,Label
+      instrAddr = reinterpret_cast<intptr_t>(cursor);
+      *reinterpret_cast<int8_t *>(cursor) = static_cast<int8_t>(0xEC);
+      cursor += sizeof(int8_t);
+      *reinterpret_cast<int8_t *>(cursor) = static_cast<int16_t>(((cg()->getEntryPointRegister() - 1) << 4 ) + 0x6 );
+      cursor += sizeof(int8_t);
+
+      // Now we need to calculate the instruction address for the resolved case, for now Just Skipping the 4 bytes
+      cursor += sizeof(int32_t);
+
+      TR::SymbolReference *helperGlueSymRef = cg()->symRefTab()->findOrCreateRuntimeHelper(callSymRef->getSymbol()->castToMethodSymbol()->isStatic() ? TR_S390interpretedUnresolvedStaticCallGlueRX : TR_S390interpretedUnresolvedSpecialCallGlueRX,
+                                                                                             false, false, false);
+      TR::GlobalFunctionCallData dataDestination(helperGlueSymRef, callNode, cursor, cg(), self());
+
+      cursor = cg()->getObjFmt()->encodeGlobalFunctionCall(dataDestination);
+      // Calling the Encode Global FunctionCall and now update the 
+      
```

Comment:
This comment looks incomplete.

Reply:
Comment was not needed, removed it in https://github.com/eclipse/openj9/compare/9a9d9b37daeb983c8d4e295fcc785afeecde5d78..de725983cd06ff96468a76b30321d3f8bb48caea